civihr_employee_portal/civihr_employee_portal.module

```diff
@@ -100,7 +100,128 @@ function _rebuild_absence_entitlement_view() {
+    }
+  }
+}
+
+/**
+ * Rebuilds Documents view.
+ */
+function _rebuild_documents_view() {
+  $extensions = CRM_Core_PseudoConstant::getExtensions();
+  if (!empty($extensions['uk.co.compucorp.civicrm.tasksassignments'])) {
+    $activityTypes = _get_activity_type_ids_by_component('CiviDocument');
+    if (!empty($activityTypes)) {
+      db_query('DROP VIEW IF EXISTS documents');
+      db_query('CREATE VIEW documents AS ' . _get_tasks_documents_sql_query($activityTypes));
+
+      // Set to false so it will not rebuild until not asked to rebuild
+      variable_set('rebuild_documents_view', 'FALSE');
+    }
+  }
+}
+
+/**
+ * Return an array containg activity type IDs of given component.
+ *
+ * @param string $component
+ * @return array
+ * @throws API_Exception
+ */
+function _get_activity_type_ids_by_component($component) {
+  $civi_settings = parse_url(CIVICRM_DSN);
+  $civi_db_name = trim($civi_settings['path'], '/');
+  $typeIds = array();
+  $optionGroup = civicrm_api3('OptionGroup', 'get', array(
+    'sequential' => 1,
+    'name' => "activity_type",
+  ));
+
+  if (!isset($optionGroup['id'])) {
+    throw new API_Exception(ts("Cannot find OptionGroup with 'name' = 'activity_type'."));
+  }
+
+  $componentQuery = "SELECT * FROM {$civi_db_name}.civicrm_component WHERE name = %1";
+  $componentParams = array(
+    1 => array($component, 'String'),
+  );
+  $componentResult = CRM_Core_DAO::executeQuery($componentQuery, $componentParams);
+  if ($componentResult->fetch())
+  {
+    $componentId = $componentResult->id;
+    $result = civicrm_api3('OptionValue', 'get', array(
+      'sequential' => 0,
+      'option_group_id' => $optionGroup['id'],
+      'component_id' => $componentId,
+    ));
+    if (!empty($result['values'])) {
+      foreach ($result['values'] as $value) {
+        $typeIds[] = $value['value'];
+      }
+    }
+  }
+
+  return $typeIds;
+}
+
+/**
+ * Return a string containing SQL query to select Activity data required
+ * for create SQL View for Tasks / Documents.
+ * $activityTypes array should contain a set activity_type_id pointing to
+ * Task types or Document types.
+ *
+ * @param array $activityTypes
+ * @return string
+ */
+function _get_tasks_documents_sql_query(array $activityTypes) {
+  $civi_settings = parse_url(CIVICRM_DSN);
+  $civi_db_name = trim($civi_settings['path'], '/');
+  return "SELECT a.id, a.activity_type_id, a.subject, a.activity_date_time, acustom.expire_date, a.details, a.status_id, a.is_deleted, ca.case_id, COUNT(ef.id) AS file_count, acs.contact_id AS source_contact_id, contact_source.sort_name AS source_contact_name, act.contact_id AS target_contact_id, contact_target.sort_name AS target_contact_name, aca.contact_id AS assignee_contact_id, contact_assignee.sort_name AS assignee_contact_name
+            FROM {$civi_db_name}.civicrm_activity a
+            LEFT JOIN {$civi_db_name}.civicrm_value_activity_custom_fields_11 acustom ON acustom.entity_id = a.id
```

Comment:
is it really save to depend on this civicrm_value_activity_custom_fields_11 ... maybe a call to civicrm API to get the custom field ID via it's name or label is a better approach .. or maybe not that easy ?! .. just thinking out loud.